### PR TITLE
Abstract impersonate button functionality for userable type

### PR DESCRIPTION
**Pull request by m-Peter**
:warning: DO NOT MERGE yet, it still needs a related PR from backend

### CODE REVIEW

File: src/javascripts/ng-admin/Crud/CrudModule.js [hunk 1]
```diff
@@ -101,6 +101,7 @@ CrudModule.directive('maSubmitButton', require('./button/maSubmitButton'));
 CrudModule.directive('maViewBatchActions', require('./button/maViewBatchActions'));
 CrudModule.directive('maImpersonateHostButton', require('./button/maImpersonateHostButton'));
 CrudModule.directive('maImpersonateTaskerButton', require('./button/maImpersonateTaskerButton'));
```
Comment (FelixMalfait) [inline src/javascripts/ng-admin/Crud/CrudModule.js]: Remove maImpersonateTaskerButton // maImpersonateHostButton ?

File: src/javascripts/ng-admin/Crud/button/maImpersonateUserableButton.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+            scope.isDeleted = (scope.entry().values.deleted_at) ? true : false;
+        },
+        template:
+`<div ng-if="isDeleted == false" class="btn-group" uib-dropdown is-open="status.isopen">
+      <button id="single-button" type="button" class="btn btn-default" uib-dropdown-toggle ng-disabled="disabled">
+        <span class="glyphicon glyphicon-random"></span>  Impersonate <span class="caret"></span>
+      </button>
+      <ul class="dropdown-menu" uib-dropdown-menu role="menu" aria-labelledby="single-button">
+        <li role="menuitem"><a href="{{ targetdomain }}auth/impersonate/{{ entry().values.id }}?as={{ token }}" target="_blank"><span class="hidden-xs" translate="Desktop App" title="Go to {{ domain }} under this {{ userable }} identity"></span></a></li>
```
Comment (FelixMalfait) [inline src/javascripts/ng-admin/Crud/button/maImpersonateUserableButton.js]: Phrase is probably a bit too long, maybe keep the previous wording? Or reduce the length a bit :)

**Comment by FelixMalfait** [comment]
Beside small comments, looks good to me!
(Merge conflict, you have to merge master and make build again)

File: src/javascripts/ng-admin/Crud/CrudModule.js (hunk 1 above)
Comment (m-Peter) [reply src/javascripts/ng-admin/Crud/CrudModule.js]: Good point!

**Pull request merged**